usage-guides/api-keys.md

```diff
@@ -149,42 +149,61 @@ After following these steps, your balances in the dashboard will be updated incl
 
 ### Monerium
 
+You can integrate Monerium with rotki to import your transaction data. This feature is only available for premium users.
+
 As long as you provide Monerium credentials, all Monerium transactions on Mainnet, Polygon, and Gnosis chain will be decorated with bank data (or chain bridging data), which you can also see in the Monerium app.
 
 1. For bank transfers from/to your address, you will be able to see the destination/source IBAN along with the memo of the transfer.
 2. For automatic EURe bridging between EVM chains, you will see amounts, from/to EVM chains along with from/to address.
 
 ![Monerium decorated transactions](/images/monerium_transactions.png)
 
-> **Note**: Monerium API can only use basic authentication at the moment. We are in contact with their team and they will add API keys eventually. But for now, they only have basic authentication, which means your username and password are stored in rotki (encrypted). Take necessary security precautions and don't use unless you understand the potential risks.
+#### Adding Monerium OAuth Token
+
+![Monerium get access token](/images/monerium_get_access_token.png)
+![Monerium OAuth](/images/monerium_oauth.png)
+
+To connect your Monerium account:
+
+1. **Get Access Token:**
+   - In rotki, navigate to `API Keys → External Services → Monerium`.
+   - Click the **Get access token** button.
+
+2. **Complete OAuth Flow:**
+   - You will be redirected to rotki.com to connect via Monerium OAuth.
+   - Follow the authentication flow on the Monerium OAuth page.
+
+3. **Return to rotki:**
+   - **Standard mode**: You will be automatically redirected back to the app after completing the OAuth flow.
+   - **Docker mode**: You will need to manually copy the access token and refresh token from the OAuth page and input them into rotki.
+
+4. **Redecode Transactions:**
+   - After successfully adding your Monerium credentials, you need to redecode the Monerium transactions in the history events section to see the updated bank data.
```

Comment:
maybe include a link to the part of the docs that explains how to redecode monerium transactions.